src/background/services/openPayments.ts

```diff
@@ -330,25 +338,41 @@ export class OpenPaymentsService {
       throw new Error('Expected finalized grant. Received non-finalized grant.')
     }
 
-    const token = {
-      value: continuation.access_token.value,
-      manage: continuation.access_token.manage
+    const grantDetails: GrantDetails = {
+      type: recurring ? 'recurring' : 'one-time',
+      amount: transformedAmount as Required<WalletAmount>,
+      accessToken: {
+        value: continuation.access_token.value,
+        manageUrl: continuation.access_token.manage
+      },
+      continue: {
+        accessToken: continuation.continue.access_token.value,
+        url: continuation.continue.uri
+      }
     }
 
-    await this.storage.set({
+    const data = {
       walletAddress,
       rateOfPay,
       minRateOfPay,
       maxRateOfPay,
-      amount: transformedAmount,
-      token,
-      grant: {
-        accessToken: continuation.continue.access_token.value,
-        continueUri: continuation.continue.uri
-      },
       connected: true
-    })
-    this.token = token
+    }
+    if (grantDetails.type === 'recurring') {
+      await this.storage.set({
+        ...data,
+        recurringGrant: grantDetails,
+        recurringGrantRemainingBalance: amount
+      })
+    } else {
+      await this.storage.set({
+        ...data,
+        oneTimeGrant: grantDetails,
+        oneTimeGrantRemainingBalance: amount
+      })
+    }
+    this.grant = grantDetails
+    this.token = this.grant.accessToken
```

Comment:
Nit: Keeping `this.token` to avoid unrelated changes (we can use this.grant.accessToken` everywhere now).